can/interfaces/socketcan/socketcan_native.py

```diff
@@ -338,6 +368,9 @@ def captureMessage(sock):
     is_extended_frame_format = bool(can_id & 0x80000000)
     is_remote_transmission_request = bool(can_id & 0x40000000)
     is_error_frame = bool(can_id & 0x20000000)
+    is_fd = len(cf) > 16
```

Comment:
Reading through the [can kernel docs](https://www.kernel.org/doc/Documentation/networking/can.txt) we could probably be more explicit here - checking the len(cf) is exactly what we expect.

>   Example:
>    [ remember: CANFD_MTU == sizeof(struct canfd_frame) ]

```
    struct canfd_frame cfd;

    nbytes = read(s, &cfd, CANFD_MTU);

    if (nbytes == CANFD_MTU) {
            printf("got CAN FD frame with length %d\n", cfd.len);
	    /* cfd.flags contains valid data */
    } else if (nbytes == CAN_MTU) {
            printf("got legacy CAN frame with length %d\n", cfd.len);
	    /* cfd.flags is undefined */
    } else {
            fprintf(stderr, "read: invalid CAN(FD) frame\n");
            return 1;
    }
```